File: 151002/Saprin/RV/discussion/src/main/java/by/bsuir/discussion/service/comment/ICommentService.java
package by.bsuir.discussion.service.comment;

import by.bsuir.discussion.dto.CommentRequestTo;
import by.bsuir.discussion.dto.CommentResponseTo;
import by.bsuir.discussion.exception.DuplicateEntityException;
import by.bsuir.discussion.exception.EntititesNotFoundException;
import by.bsuir.discussion.exception.EntityNotFoundException;

import java.math.BigInteger;
import java.util.List;

public interface ICommentService {
    List<CommentResponseTo> getComments() throws EntititesNotFoundException;

    CommentResponseTo addComment(CommentRequestTo comment) throws EntityNotFoundException, DuplicateEntityException;
    void deleteComment(BigInteger id) throws EntityNotFoundException;
    CommentResponseTo updateComment(CommentRequestTo comment) throws EntityNotFoundException, DuplicateEntityException;
    CommentResponseTo getCommentById(BigInteger id) throws EntityNotFoundException;
}

File: 151003/Mitrofanov/Discussion/CommentEntity/Interface/ICommentService.cs
using Discussion.Common.Interface;
using Discussion.CommentEntity.Dto;

namespace Discussion.CommentEntity.Interface
{
    public interface ICommentService : ICrudService<Comment, CommentRequestTO, CommentResponseTO>
    {
        Task<IList<Comment>> GetByStoryID(int storyId);
    }
}
